### PR TITLE
Bug 1126685 - use attributechanged to monitor v2 BT stat +autoland

### PR DESCRIPTION
**Pull request by gasolin**

### CODE REVIEW

**Comment by try-server-hook** [comment]
[gasolin gasolin (gasolin)](https://github.com/gasolin) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=2121ad056d3d)

**Comment by try-server-hook** [comment]
[gasolin gasolin (gasolin)](https://github.com/gasolin) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=2128c4b007b7)

File: dev_apps/uitest/js/HW/bluetooth.js [hunk 1]
```diff
@@ -10,22 +10,23 @@ function bluetoothTest() {
...
         address.innerHTML = 'unknown';
       }
       else {
         state.innerHTML = 'On';
-        btHelper.getAddress(function bt_gotAddress(addr) {
-          if (isBluetoothEnabled) {
-            address.innerHTML = addr;
-          }
-        });
+        if (!btHelper.v2) {
```
Comment (gasolin) [inline dev_apps/uitest/js/HW/bluetooth.js]: only v1 use btHelper.getAddress

File: dev_apps/uitest/js/HW/bluetooth.js [hunk 2]
```diff
@@ -45,16 +46,49 @@ function bluetoothTest() {
 
   if (btHelper.v2) {
     version.textContent = "v2";
```
Comment (gasolin) [inline dev_apps/uitest/js/HW/bluetooth.js]: v2 will listen to APIv2 'attributechanged' directly to change state and address

File: dev_apps/uitest/js/HW/bluetooth.js [hunk 3]
```diff
@@ -45,16 +46,49 @@ function bluetoothTest() {
...
-    bluetooth.onadapteradded = function onAdapterAdded(evt) {
-      update();
-    }
+    bluetooth.addEventListener('attributechanged', (evt) => {
+      for (var i in evt.attrs) {
+        switch (evt.attrs[i]) {
+          case 'defaultAdapter':
+            var adapter = bluetooth.defaultAdapter;
+            if (adapter.address) { // for default On case
```
Comment (gasolin) [inline dev_apps/uitest/js/HW/bluetooth.js]: the BT address will already got when BT already ON, so the 'attributechanged' wont get the event

File: dev_apps/uitest/js/HW/bluetooth.js [hunk 4]
```diff
@@ -45,16 +46,49 @@ function bluetoothTest() {
...
+              address.innerHTML = adapter.address;
+            }
+            adapter.addEventListener('attributechanged', (evt) => {
+              for (var i in evt.attrs) {
+                switch (evt.attrs[i]) {
+                  case 'state':
+                    if (adapter.state === 'enabled') {
+                      switchButton.checked = true;
```
Comment (gasolin) [inline dev_apps/uitest/js/HW/bluetooth.js]: make sure the button state is right when swiping from other apps

File: dev_apps/uitest/js/HW/bluetooth.js [hunk 5]
```diff
@@ -45,16 +46,49 @@ function bluetoothTest() {
...
+            break;
+          default:
+            break;
+        }
+      }
+    });
   } else {
     version.textContent = "v1";
     bluetooth.addEventListener('adapteradded', update);
+    bluetooth.addEventListener('enabled', update);
```
Comment (gasolin) [inline dev_apps/uitest/js/HW/bluetooth.js]: wrap enabled/disabled listener for v1 only

File: shared/js/bluetooth_helper.js [hunk 6]
```diff
@@ -196,6 +196,11 @@ var BluetoothHelper = function() {
     },
 
     getAddress: function(cb) {
+      if (_v2) {
```
Comment (gasolin) [inline shared/js/bluetooth_helper.js]: as discussed, we will not support getAddress in v2

**Comment by ian-liu** [comment]
:100:

**Pull request merged**